# 가설검정 연습문제 (2) 
## 문제 1
어떤 화학약품의 제조에 상표가 다른 두 종류의 원료가 사용되고 있다.

각 원료에서 주성분 A의 함량은 다음과 같다. 단, 함량은 정규분포를 따른다고 가정한다.

이 두 원료의 주성분 A의 함량이 다른지를 분석하라.

https://data.hossam.kr/E02/material.xlsx

>> 독립표본 T검정



In [11]:
import pandas as pd 
import scipy.stats  as stats

from matplotlib import pyplot as plt
import seaborn as sb    
import sys

In [12]:
df = pd.read_excel("https://data.hossam.kr/E02/material.xlsx")
df

상표1   상표2
0   80.4  80.1
1   78.2  81.2
2   80.1  79.5
3   77.1  78.0
4   79.6  76.1
5   80.4  77.0
6   81.6  80.1
7   79.9  79.9
8   84.4  78.8
9   80.9  80.8
10  83.1   NaN

귀무가설($H_0$) :상표1과 상표2에서 주성분A의 함량은 같다

대립가설($H_1$) :상표1과 상표2에서 주성분A의 함량은 같지않다

In [14]:
brand1 = df['상표1']
brand2 = df['상표2'].dropna(axis=0)

In [17]:
stats.ttest_ind(brand1,brand2,alternative="two-sided")

TtestResult(statistic=1.676025364631745, pvalue=0.11011138184228597, df=19.0)

#### 결과해석

유의수준 0.05에서 독립표본 T검정 결과, 검정통계량은 1.6760, p값은 0.11011이 나온것을 확인했음. 

유의수준 0.05에서 p값이 0.05보다 크므로 귀무가설을 채택한다. 즉, 상표1과 상표2에서의 주성분A의 함령은 같다고 할 수 있다 .

## 문제 2

특정 피임약이 사용자의 혈압을 저하시키는지 조사하고자 한다.

이를 위해 부인 15명을 대상으로 평상시 혈압을 측정한 뒤, 이들에게 이 피임약을 일정 기간 복용하게 한 후 이들의 혈압을 다시 측정한 결과를 기록했다.

얻어진 데이터는 다음과 같다. 피임약 복용이 혈압에 영향을 주는지 분석하라

>> 쌍체표본 T검정

In [18]:
df = pd.read_excel("https://data.hossam.kr/E02/blood_pressure.xlsx")
df

복용전  복용후
0    70   68
1    80   72
2    72   62
3    76   70
4    76   58
5    76   66
6    72   68
7    78   52
8    82   64
9    64   72
10   74   74
11   92   60
12   74   74
13   68   72
14   84   74

In [19]:
stats.ttest_rel(df['복용전'],df['복용후'],alternative='two-sided')

TtestResult(statistic=3.105360487466109, pvalue=0.007749436106654119, df=14)

#### 결과해석

유의수준 0.05에서 쌍체표본 T검정 결과, 검정통계량은 3.10, p값은 0.007이 나온것을 확인했음. 

유의수준 0.05에서 p값이 0.05보다 작으므로 귀무가설을 기각한다. 즉, 피암약 복용 전과 후 혈압이 다르기 때문에 피임약 복용은 혈압에 영향을 준다

## 문제 3
고등학교 육상선수에게 체중감량을 시키면 달리기에 어떤 변화가 일어나는지 조사하기 위 하여, 어떤 고등학교 육상선수 10명에게 감량훈련(2~4kg 정도)을 시킨 후 100m 달리기를 실시하여 다음의 데이터를 얻었다.

감량 전과 후에 달리기 속도에 차이가 있는지 분석하라

>> 쌍체표본 T검정

In [20]:
df = pd.read_excel("https://data.hossam.kr/E02/runner_diet.xlsx")
df

감량전   감량후    
0  14.5  14.3 NaN
1  13.4  13.4 NaN
2  13.0  13.1 NaN
3  14.5  13.9 NaN
4  13.3  13.1 NaN
5  14.2  13.9 NaN
6  13.6  13.6 NaN
7  13.9  14.0 NaN
8  13.2  12.8 NaN
9  13.0  12.8 NaN

In [21]:
stats.ttest_rel(df['감량전'],df['감량후'],alternative='two-sided')


TtestResult(statistic=2.375306807446375, pvalue=0.04154604923393679, df=9)

#### 결과해석

유의수준 0.05에서 쌍체표본 T검정 결과, 검정통계량은 2.37, p값은 0.0415가 나온것을 확인했음. 

유의수준 0.05에서 p값이 0.05보다 작으므로 귀무가설을 기각한다. 즉, 감량 전과 후 달리기 속도가 다르기 떄문에 감량훈련이 달리기 속도에 영향을 준다고 할수있음,

## 문제 4
고등학교 레슬링 선수에게 체중감량을 시키면 체력에 어떤 변화가 일어나는지 조사하기 위하여, 어느 고등학교의 레슬링 선수 12명을 표본추출하여 감량시킨 후 다음의 체력 데이터를 얻었다.

감량 전과 감량 후 간에 차이가 있는가의 대응비교를 악력(우), 윗몸일으키기, 턱걸이, 100m 달리기에 대하여 실시하고 분석하라.

>> 쌍체표본 T검정

In [22]:
df = pd.read_excel("https://data.hossam.kr/E02/wrestler_diet.xlsx")
df

이름  감량전 악력  감량후 악력  감량전 윗몸일으키기  감량후 윗몸일으키기  감량전 턱걸이  감량후 턱걸이  \
0   B.S      43      41          35          41       25       29   
1   P.J      42      41          40          44       25       29   
2   G.G      52      50          36          41       22       23   
3   A.L      53      52          38          42       10       15   
4   I.H      44      40          36          42       15       19   
5   B.Y      46      47          39          44       17       19   
6   I.K      48      47          35          42       24       26   
7   K.B      58      55          34          36       17       18   
8   K.J      50      50          33          35       17       20   
9   S.S      48      50          39          45       21       25   
10  S.P      56      48          37          37       31       35   
11  S.J      43      41          33          37       15       22   

    감량전 100m 달리기  감량후 100m 달리기  
0           14.5          14.5  
1           13.4          13.4  
2           13.5          13.5  
3           13.9          13.9  
4           13.4          13.4  
5           13.9          13.9  
6           13.6          13.6  
7           13.9          13.9  
8           13.1          13.1  
9           13.8          13.8  
10          13.7          13.7  
11          14.9          14.9

### 악력 대응 비교

In [23]:
stats.ttest_rel(df['감량전 악력'],df['감량후 악력'],alternative='two-sided')


TtestResult(statistic=2.3654081485744824, pvalue=0.037454446305019644, df=11)

###  악력 결과 해석

유의수준 0.05에서 쌍체표본 T검정 결과, 검정통계량은 2.3654, p값은 0.03745가 나온것을 확인했음. 

유의수준 0.05에서 p값이 0.05보다 작으므로 귀무가설을 기각한다. 즉, 감량 전과 후 악략이 다르기 떄문에 감량훈련이 악력에 영향을 준다고 할수있음,

### 윗몸일으키기 대응비교

In [28]:
stats.ttest_rel(df['감량전 윗몸일으키기'],df['감량후 윗몸일으키기'],alternative='two-sided')

TtestResult(statistic=-7.179926973956724, pvalue=1.798236117030195e-05, df=11)

### 윗몸일으키기 결과 해석
유의수준 0.05에서 쌍체표본 T검정 결과, 검정통계량은 -7.1799, p값은 0에 가까운 값이 나온것을 확인했음. 

유의수준 0.05에서 p값이 0.05보다 작으므로 귀무가설을 기각한다. 즉, 감량 전과 후 윗몸일으키기 갯수가 다르기 떄문에 감량훈련이 윗몸일으키기 갯수에 영향을 준다고 할수있음,

### 턱걸이 대응비교

In [30]:
stats.ttest_rel(df['감량전 턱걸이'],df['감량후 턱걸이'],alternative='two-sided')

TtestResult(statistic=-6.841977654731714, pvalue=2.7937021187235204e-05, df=11)

### 턱걸이 결과 해석
유의수준 0.05에서 쌍체표본 T검정 결과, 검정통계량은 -6.8419, p값은 0에 가까운 값이 나온것을 확인했음. 

유의수준 0.05에서 p값이 0.05보다 작으므로 귀무가설을 기각한다. 즉, 감량 전과 후 턱걸이 갯수가 다르기 떄문에 감량훈련이 턱걸이 갯수에 영향을 준다고 할수있음,

### 100m 달리기 대응비교

In [35]:
stats.ttest_rel(df['감량전 100m 달리기'],df['감량후 100m 달리기'])

TtestResult(statistic=nan, pvalue=nan, df=11)

이름  감량전 악력  감량후 악력  감량전 윗몸일으키기  감량후 윗몸일으키기  감량전 턱걸이  감량후 턱걸이  \
0   B.S      43      41          35          41       25       29   
1   P.J      42      41          40          44       25       29   
2   G.G      52      50          36          41       22       23   
3   A.L      53      52          38          42       10       15   
4   I.H      44      40          36          42       15       19   
5   B.Y      46      47          39          44       17       19   
6   I.K      48      47          35          42       24       26   
7   K.B      58      55          34          36       17       18   
8   K.J      50      50          33          35       17       20   
9   S.S      48      50          39          45       21       25   
10  S.P      56      48          37          37       31       35   
11  S.J      43      41          33          37       15       22   

    감량전 100m 달리기  감량후 100m 달리기  
0           14.5          14.5  
1           13.4          13.4  
2           13.5          13.5  
3           13.9          13.9  
4           13.4          13.4  
5           13.9          13.9  
6           13.6          13.6  
7           13.9          13.9  
8           13.1          13.1  
9           13.8          13.8  
10          13.7          13.7  
11          14.9          14.9

### 100m 결과 해석
표본을 확인했을때 감량 전 후 기록에 있어서 차이가 없는 것과, 쌍체표본 t검정 결과 nan값이 나오는 것을 확인했다. 

이를 바탕으로 감량 전 후 100m달리기의 기록은 감량훈련에 관계없이 똑같거나, 다른 방식을 통해 표본을 재추출하여 검정을 재실시해야함.